File: app/src/main/java/io/ipoli/android/app/ui/calendar/CalendarEvent.java
package io.ipoli.android.app.ui.calendar;

import android.support.annotation.ColorRes;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 2/17/16.
 */
public interface CalendarEvent {
    Integer getStartMinute();

    int getDuration();

    void setStartMinute(Integer startMinute);

    @ColorRes
    int getBackgroundColor();

    @ColorRes
    int getDragBackgroundColor();

    boolean isRepeating();

    String getName();

    boolean isMostImportant();

    boolean isForChallenge();
}

File: app/src/main/java/io/ipoli/android/quest/calendar/ui/CalendarEvent.kt
package io.ipoli.android.quest.calendar.ui

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 9/2/17.
 */
interface CalendarEvent {
    var duration: Int
        get

    var startMinute: Int
        get
}